### ISSUE TITLE
[Training] No Op registered for SimplifiedLayerNormalization with domain_version of 14 when using generate_artifatcs

### ISSUE THREAD

**Issue by martinkorelic** [microsoft/onnxruntime#21827]
### Describe the issue

I am trying to generate artifacts for on device training using a custom transformer model from Huggingface, however I am running into this issue with the custom operators and I am unsure whether they are supported for training with ONNX Runtime training.

Are these operators already supported by ONNX runtime training or not?
Or perhaps I am running into another issue with the operators not being included in my build?

### To reproduce

I am using optimum exporters for onnx runtime to create the `model.onnx` from transformers which includes optimizations for several operations:
1. Have installed optimum and onnx runtime
2. `optimum-cli export onnx -m model_name --optimize O2 model_output`
3. Try to generate artifacts using the exported model. E.g.:
```py
artifacts.generate_artifacts(onnx_model,
                                requires_grad = requires_grad,
                                frozen_params = frozen_params,
                                loss = LossFunction(),
                                optimizer = artifacts.OptimType.AdamW,
                                artifact_directory = onnx_model_dir)
```

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.20.0+cpu

### PyTorch Version

2.2.0+rocm5.6

### Execution Provider

Default CPU

**Comment by carzh** [comment]
Custom operators are not supported in on-device training -- the forward ONNX graph passed to the generate_artifacts call should be a non-optimized model without custom ops. All the ops it contains should be able to be found here: [ONNX opset](https://onnx.ai/onnx/operators/index.html).

If you want to get around it, you can comment out the ONNX checker line in generate_artifacts: [this line if building from source](https://github.com/microsoft/onnxruntime/blob/ff3e8b02c3c7eaf5bc057fa4455a2c973ae97f96/orttraining/orttraining/python/training/onnxblock/blocks.py#L65), otherwise would be in your `site-packages/onnxruntime/training/onnxblock/blocks.py`

However, since you're exporting an optimized model, you will most likely run into issues with any other optimized / custom ops in your model. No gradient operators exist for those ops in on-device training.

**Comment by martinkorelic** [comment]
Thank you for answering, as I found out experimenting and mentioned by others, some optimizations contain ONNX ops and optimized ORT contrib ops, which do not implement gradient operations, effectively not allowing to generate training artifacts.

This issue can be closed now.
